api/views/registration.py

```diff
@@ -0,0 +1,52 @@
+from rest_framework import viewsets, status
+from rest_framework.response import Response
+
+from api.permissions import ReadOnlyOrTokenHasScopeOrIsAuthenticated
+from api.serializers import RegistrationFormSerializer
+from share.models import RegistrationForm
+
+
+class RegistrationFormViewSet(viewsets.ModelViewSet):
+    """View showing all registration data in the SHARE Dataset.
+
+    ## Submit Registration.
+
+    Create
+
+        Method:        POST
+        Body (JSON):   {
+                            "contact_name": "John Doe",
+                            "contact_email": "email@email.com",
+                            "contact_affiliation": "Organization affliation",
+                            "meta_tos": true,
+                            "meta_rights": true,
+                            "meta_privacy": true,
+                            "meta_sharing": true,
+                            "meta_license": true,
+                            "direct_source": true,
+                            "source_name": "Organization Name",
+                            "source_description": "Organization description.",
+                            "source_rate_limit": "(Optional) 1 request/second",
+                            "source_documentation": "(Optional)",
+                            "source_preferred_metadata_prefix": "(Optional)",
+                            "source_oai": false,
+                            "source_base_url": "(Optional)",
+                            "source_disallowed_sets": "(Optional)",
+                            "source_additional_info": "(Optional)"
+                        }
+
+        Success:       200 OK
+    """
+    permission_classes = [ReadOnlyOrTokenHasScopeOrIsAuthenticated, ]
+    serializer_class = RegistrationFormSerializer
+
+    def get_queryset(self):
+        return RegistrationForm.objects.all()
+
+    def create(self, request, *args, **kwargs):
+        serializer = RegistrationFormSerializer(data=request.data, context={'request': request})
+        if serializer.is_valid():
+            nm_instance = serializer.save()
+            return Response({'registration_id': nm_instance.id}, status=status.HTTP_202_ACCEPTED)
```

Comment:
Successful create requests should return 201 (Created).
202s signify that your initial data was accepted but there is further processing to be done.